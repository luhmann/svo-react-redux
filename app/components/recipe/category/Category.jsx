import { React, CSSModules, CSSModuleConfig } from 'lib/commonImports.js'

import styles from './Category.styl'

const categoryMapping = {
  dessert: 'Dessert & Backen'
}

const Category = ({ type }) => (
  <div styleName='category'>{ categoryMapping[type] }</div>
)

export default CSSModules(Category, styles, CSSModuleConfig)
