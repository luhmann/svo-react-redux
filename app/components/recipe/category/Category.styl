@require '~styles/variables.styl'

.category
  font-family 'Telefon'
  margin-top 0.4em
