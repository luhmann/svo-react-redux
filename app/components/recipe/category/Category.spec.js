import React from 'react'
import renderer from 'react-test-renderer'

import Category from './Category.jsx'

describe('Component: <Category />', () => {
  it('should map "breakfast"-category', () => {
    const component = renderer.create(
      <Category type="breakfast" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should map "lunch"-category', () => {
    const component = renderer.create(
      <Category type="lunch" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should map "dinner"-category', () => {
    const component = renderer.create(
      <Category type="dinner" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should map "dessert"-category', () => {
    const component = renderer.create(
      <Category type="dessert" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should map "snack"-category', () => {
    const component = renderer.create(
      <Category type="snack" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should omit an unknown category', () => {
    const component = renderer.create(
      <Category type="foo" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
