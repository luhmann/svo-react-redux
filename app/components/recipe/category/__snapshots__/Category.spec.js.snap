exports[`Component: <Category /> should map "breakfast"-category 1`] = `
<div
  className="eRHmmS">
  Frühstück
</div>
`;

exports[`Component: <Category /> should map "dessert"-category 1`] = `
<div
  className="eRHmmS">
  Dessert & Backen
</div>
`;

exports[`Component: <Category /> should map "dinner"-category 1`] = `
<div
  className="eRHmmS">
  Abendessen
</div>
`;

exports[`Component: <Category /> should map "lunch"-category 1`] = `
<div
  className="eRHmmS">
  Mittagessen
</div>
`;

exports[`Component: <Category /> should map "snack"-category 1`] = `
<div
  className="eRHmmS">
  Kleinigkeiten & Snacks
</div>
`;

exports[`Component: <Category /> should omit an unknown category 1`] = `null`;
